modules/contracts/apps/calendar/tests/integration.rs

```diff
@@ -100,58 +95,47 @@ fn setup_with_time(
     mock.set_balance(&Addr::unchecked("sender2"), coins(INITIAL_BALANCE, DENOM))?;
     mock.set_balance(&Addr::unchecked("sender"), coins(INITIAL_BALANCE, DENOM))?;
 
-    // Construct the contract interface
-    let app = AppInterface::new(APP_ID, mock.clone());
-
     // Deploy Abstract to the mock
-    let abstr_deployment = Abstract::deploy_on(mock.clone(), sender.to_string())?;
+    Abstract::deploy_on(mock.clone(), sender.to_string())?;
+
+    let client = AbstractClient::new(mock.clone())?;
 
-    abstr_deployment.ans_host.execute(
+    client.ans_host().execute(
         &abstract_core::ans_host::ExecuteMsg::UpdateAssetAddresses {
             to_add: vec![(DENOM.to_owned(), AssetInfo::native(DENOM).into())],
             to_remove: vec![],
         },
         None,
     )?;
 
-    // Create a new account to install the app onto
-    let account =
-        abstr_deployment
-            .account_factory
-            .create_default_account(GovernanceDetails::Monarchy {
-                monarch: ADMIN.to_string(),
-            })?;
+    // Create account to install app onto as well as claim namespace.
+    let publisher: Publisher<Mock> = client
+        .publisher_builder()
+        .governance_details(GovernanceDetails::Monarchy {
+            monarch: ADMIN.to_owned(),
+        })
+        .namespace("my-namespace")
+        .build()?;
 
-    // claim the namespace so app can be deployed
-    abstr_deployment
-        .version_control
-        .claim_namespace(AccountId::local(1), "my-namespace".to_string())?;
+    publisher.deploy_app::<AppInterface<Mock>>()?;
```

Comment:
Since it's a publisher, should `deploy` -> `publish`?
```suggestion
    publisher.publish_app::<AppInterface<Mock>>()?;
```

Reply:
Should we also change `deploy_adapter` to `publish_adapter`?

Reply:
Not sure. It both instantiates and registers them. Could that cause confusion? Should `publish` only do the publishing part and not instantiate it?